Feature: Accept custom query for ChromaDB RAG workflow

aryanyk opened the pull request:
### Summary
- Added support for passing a custom query string via CLI for the
  `DJ_chromadb_gen_embedding` task using `click`.
- Default query remains unchanged to preserve backward compatibility.

### Motivation
This resolves #153 by allowing users to provide custom query text
without modifying the script.

### How I Tested

pixi run DJ_chromadb_gen_embedding
pixi run DJ_chromadb_gen_embedding -q "My laptop screen cracked"

Output : 
<img width="1536" height="678" alt="image" src="https://github.com/user-attachments/assets/7f29e73f-7835-45c3-b923-ee7049ae926b" />

sayantikabanik commented on analytics_framework/rag_system/setup_chroma_vector_db.py:
```diff
@@ -58,18 +59,49 @@ def chromadb_processing(documents, ids, db_collection_name, what_to_query, resul
+@click.option(
+    "--query",
+    "-q",
+    default="My macbook broke in an accident",
```
The default fallback is great. Could we have a message stating the same?

Example: 

`defaulting to "...." since no query provided`

sayantikabanik commented on analytics_framework/rag_system/setup_chroma_vector_db.py:
```diff
@@ -58,18 +59,49 @@ def chromadb_processing(documents, ids, db_collection_name, what_to_query, resul
+@click.option(
+    "--n-results",
+    "-n",
+    "n_results",
```
Considering only one shorthand would be best (`-n` suffices)

sayantikabanik commented on analytics_framework/rag_system/setup_chroma_vector_db.py:
```diff
@@ -58,18 +59,49 @@ def chromadb_processing(documents, ids, db_collection_name, what_to_query, resul
+    Populate a ChromaDB collection from YouTube comments and run a query.
+
+    If no query is provided via CLI, a default example query is used.
+    """
```
Thanks for the additional thoughts on the docstrings. 
Format below is recommended and unifies across the framework

```python
def function_name(arg1: type, arg2: type) -> return_type:
    """
    Brief description of what the function does.

    Args:
        arg1 (type): Description of the argument.
        arg2 (type): Description of the argument.

    Returns:
        return_type: Description of what is returned.
    """
```

sayantikabanik reviewed (changes requested):
Thanks, @aryanyk for the efforts! The PR looks good 

I left a few comments. Let me know if you have any follow-up questions.

aryanyk commented:
Thanks for the review and helpful suggestions, @sayantikabanik!

I’ve made the requested updates:

- Simplified the CLI to use only shorthand flags (-q, -n)
- Added a message when falling back to the default query
- Aligned the docstring format with the framework standard
- Cleaned up the CLI output for clarity

I’ve tested both default and custom query flows locally.
Please let me know if anything else needs adjustment.
